### ISSUE TITLE
Persist getters

### ISSUE THREAD

**Issue by cinderisles** [prazdevs/pinia-plugin-persistedstate#222]
### Clear and concise description of the problem

It would be nice if getters could also be persisted. I am trying to save a snapshot of a store to sessionStorage, but if getters cannot be saved, then I must find an alternative to this library

[This line](https://github.com/prazdevs/pinia-plugin-persistedstate/blob/main/packages/plugin/src/plugin.ts#L113) suggests that this library does not support getters, as does [this reproduction](https://codesandbox.io/s/heuristic-smoke-xlf8ld?file=/src/store.js).

### Suggested solution

I'm not sure if this is a limitation of Pinia, as there is no `$getters` property, like there is with `$state`, but perhaps a workaround could be found.

Pinia's `storeToRefs` will return getters, so perhaps this could be used instead of `store.$state`. 

Also consider using Vue's `watch()` instead of `store.$subscribe` because `$subscribe` will not include getters.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/prazdevs/pinia-plugin-persistedstate/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/prazdevs/pinia-plugin-persistedstate/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by prazdevs** [comment]
Hi :) Because getters are calculated values (as in computed), usually based on other state parts, they shouldn't really be stored. Their calculation is optimized by Vue/Pinia.

Also we make use of `store.$subscribe` to be as close as possible to Pinia itself without overhead, or performance issues that new reactivity such as watch would imply.

tl;dr: it's a design choice to only persist state, as the name suggest `persistedstate`, and I'd say that if you need to persist a getter, maybe a getter wasn't the best solution at first? :)

**Comment by cinderisles** [comment]
sure, it makes sense that in most cases, the computed properties can be recomputed after hydrating the store.

maybe my use case is just too specific, but my issue is that I have two SPAs bundled together (one is old AngularJS, the other, Vue 3)

They don't share code because one uses Webpack 4 and the other uses Vite, and I decided it would be a pain to get code sharing working (which may be easier than I think, not sure. maybe a shared library would be better)

The simplest solution I could come up with is to share state over session storage. Sometimes, the Angular app might need access to a computed property in the Vue app's pinia store, which is why I would need persisted getters

But maybe just syncing through session storage is not the best way

---

i'm not necessarily asking for help with this angularjs + vue nonsense im working on, but just wanted to give some more context to this issue :)
maybe i should just find a different approach

**Comment by prazdevs** [comment]
Having worked with microfrontends, if the 2 apps are in the same window, local/session storage wont notify of changes in a same window, and sync will be broken :)
So I wouldnt use store for that reason.  I'll close this issue for now :)

**Comment by cinderisles** [comment]
true but as a hack you can poll sessionStorage with setInterval. It's performant enough for migrating away from AngularJS. The getters were just the missing piece. I'll just have to persist those manually if there's no plans with this library. Thanks for the feedback
